### ISSUE TITLE
input trim

### ISSUE THREAD

**Issue by basaran** [pablo-abc/felte#70]
Hello,

Does felte auto trim input values?

**Comment by basaran** [comment]
Hello again, related to my question above, I'm trying to come up with a best practices approach to transform the data and update the input fields at the same time.

Typical use case would be to capitalize certain input fields. At the moment, I can do a directive like `use:capitalize` and update the input value as the user types in and blurs out. If I do this, then I can't use felte's `initialValues` together with `use`. Felte overwrites the use:directive on initial page load with the default values. I would also have to attach event handlers to each input field.

I tried to use the transform option inside felte, but that only updates the $data value, it doesn't reflect the change back to DOM.

I tried to use the $isSubmitting hook together with `setField`. This updates the DOM at the first submit, but it doesn't update the `$data` value.

Is this the expected behaviour for `setField`?  What would be the right way to transform data, and update DOM at the same time?
